test/e2e/tests/test_notebook_instance.py

```diff
@@ -0,0 +1,135 @@
+        resource_plural=RESOURCE_PLURAL,
+        resource_name=resource_name,
+        spec_file=RESOURCE_SPEC_FILE,
+        replacements=replacements,
+    )
+
+    assert resource is not None
+    assert k8s.get_resource_arn(resource) is not None
+
+    yield (reference, resource, spec)
+
+
+def get_notebook_instance(notebook_instance_name: str):
+    try:
+        desired_notebook_instance = sagemaker_client().describe_notebook_instance(NotebookInstanceName=notebook_instance_name)
+        return desired_notebook_instance
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance with the name {notebook_instance_name}. Error {error}"
+        )
+        return None
+
+def get_notebook_instance_sagemaker_status(notebook_instance_name: str):
+    notebook_instance = get_notebook_instance(notebook_instance_name)
+    assert notebook_instance is not None
+    return notebook_instance["NotebookInstanceStatus"]
+
+def get_notebook_instance_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert resource is not None
+    assert "notebookInstanceStatus" in resource["status"]
+    return resource["status"]["notebookInstanceStatus"]
+
+@pytest.mark.canary
+@service_marker
+class TestNotebookInstance:
+    def _wait_resource_notebook_status(self,reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30):
+        return wait_for_status(expected_status,
+            wait_periods,
+            period_length,
+            get_notebook_instance_resource_status,
+            reference
+        )
+    def _wait_sagemaker_notebook_status(self,notebook_instance_name: str,
+        expected_status: str,
+        wait_periods: int = 30,
+        period_length: int = 30):
+        return wait_for_status(expected_status,
+            wait_periods,
+            period_length,
+            get_notebook_instance_sagemaker_status,
+            notebook_instance_name
+        )
+    def _assert_notebook_status_in_sync(self, notebook_instance_name, reference, expected_status):
+        assert(
+            self._wait_sagemaker_notebook_status(notebook_instance_name,expected_status,wait_periods=60,period_length=15)
+        == self._wait_resource_notebook_status(reference,expected_status,wait_periods=60,period_length=15)
+        == expected_status
+        )
+    def testCreateAndDelete(self,notebook_instance):
+        (reference, resource, spec) = notebook_instance
+        assert k8s.get_resource_exists(reference)
+
+        notebook_instance_name = resource["spec"].get("notebookInstanceName",None)
+        assert notebook_instance_name is not None
+
+        notebook_description = get_notebook_instance(notebook_instance_name)
+        assert notebook_description["NotebookInstanceStatus"] == "Pending"
```

Comment:
can we also do self._assert_notebook_status_in_sync(notebook_instance_name,reference,"Pending") or will this cause flakyness as the time in pending may be too short?